### PR TITLE
A simple operator to post messages to a Slack channel

### PR DESCRIPTION
**Pull request by syvineckruyk**
SlackAPIOperator : A base operator for the Slack API
SlackAPIPostOperator : A simple operator to post messages to a Slack channel

### CODE REVIEW

**Comment by syvineckruyk** [comment]
rebased merge conflicts resolved

File: airflow/operators/slack_operator.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+
+class SlackAPIOperator(BaseOperator):
+    """
+    Base Slack Operator
+    The SlackAPIPostOperator is derived from this operator.
+    In the future additional Slack API Operators will be derived from this class as well
+
+    :param token: Slack api token
+    :type token: String
```
Comment (mistercrunch) [inline airflow/operators/slack_operator.py:12]: `NameError: name 'String' is not defined` -> str

File: docs/code.rst [hunk 2]
```diff
@@ -63,6 +63,8 @@ Operator API
         PythonOperator,
         S3KeySensor,
         S3ToHiveTransfer,
+        SlackAPIOperator,
```
Comment (mistercrunch) [inline docs/code.rst:66]: I think sphinx won't be able to find this one if it's not added to `operators/__init__.py`

**Comment by mistercrunch** [comment]
I'm surprised that there isn't some sort of authentication required. I'll give it a spin tomorrow and see if it work against our Slack instance!

**Comment by mistercrunch** [comment]
I've been trying to get our internal token to try this but haven't found it...

**Comment by syvineckruyk** [comment]
https://api.slack.com/web

**Comment by mistercrunch** [comment]
OMG OMG OMG!

This is awesome! THE PR OF THE MONTH AWARD HAS BEEN AWARDED TO syvineckruyk!

screen shot 2015-08-06 at 4 15 40 pm

**Comment by mistercrunch** [comment]
Merged through other means. Thanks man!

**Comment by syvineckruyk** [comment]
@mistercrunch did your merge actually build ? I had not yet made the changes you requested above ?

**Comment by syvineckruyk** [comment]
@mistercrunch glad you liked it ... this was a small preview ... working on a slackbot now for airflow that will actually sit in channels as a user ... broadcast messages from airflow to whatever channels it's in and take commands as input .. im currently trying to add all the functionality of the CLI.

**Pull request closed without merging**